pkg/webhook/webhook.go

```diff
@@ -429,6 +430,26 @@ func createVolPatch(patch []jsonPatchOperation, hugepageResourceList []hugepageR
 	return patch
 }
 
+func updateNodeSelector(existing map[string]string, desired map[string]string, patch []jsonPatchOperation) []jsonPatchOperation {
+	newExistingMap := make(map[string]string)
+	if existing != nil {
+		for k, v := range existing {
+			newExistingMap[k] = v
+		}
+	}
+	if desired != nil {
+		for k, v := range desired {
+			newExistingMap[k] = v
+		}
+	}
+	patch = append(patch, jsonPatchOperation{
```

Comment:
I assume we don't need to patch nodeSelector path when pod manifest doesn't have nodeSelector defined?  Like how the `patchEmptyResources` function does.

Reply:
correct, now updated the code not to add nodeSelector path when labels map is empty.